lib/matplotlib/widgets.py

```diff
@@ -2595,3 +2598,190 @@ def onmove(self, event):
             self.canvas.blit(self.ax.bbox)
         else:
             self.canvas.draw_idle()
+
+
+LABELS_CMAP = ListedColormap(['white', 'red', 'dodgerblue', 'gold',
+                              'greenyellow', 'blueviolet'])
+
+
+class Painter(_SelectorWidget):
+    """
+    Paint regions on an axes.
+
+    For the widget to remain responsive you must keep a reference to it.
+
+    Example usage::
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from matplotlib.widgets import Painter
+
+        data = np.random.rand(100, 2)
+        data[:, 1] *= 2
+
+        fig, ax = plt.subplots()
+        pts = ax.scatter(data[:, 0], data[:, 1], s=80)
+
+        def test(x, y):
+            print(x, y)
+
+        p = Painter(ax, test)
+        plt.show()
+    """
+
+    def __init__(self, ax, on_select=None, overlay_props=None,
+                 cursor_props=None, radius=5, cmap=LABELS_CMAP,
+                 useblit=True, button=None, state_modifier_keys=None):
+        """
+        Parameters:
+
+        *ax* : :class:`~matplotlib.axes.Axes`
+            The parent axes for the widget
+        *on_select* : function
+           A callback for when a region is painted. Called with the
+           (x, y) coordinates of the region.
+        *overlay_props* : dict
+           The properties to apply to the overlay.
+        *cursor_props* : ditc
+            - The properties to apply to the cursor.
+        *radius* : int
+           - The radius of the cursor in pixels.
+        *cmap* : :class:~matplotlib.colorls.ListedColormap`
+            - The colormap to use for the cursors.
```

Comment:
How are the alternate colours used? I can't get the example to trigger anything other than a red square.